### PR TITLE
Long paths support (Windows)

### LINKED ISSUE #168: Support long paths (>= 260 chars) for net472 target runtime in Windows
See also [Why does the 260 character path length limit exist in Windows?](https://stackoverflow.com/q/1880321/1046374)

### PR DESCRIPTION
**Pull request by KvanTTT**
fixes #168

### CODE REVIEW

File: Sources/PT.PM.Common/CommonUtils.cs [hunk 1]
```diff
@@ -43,39 +43,29 @@ public static bool TryConvertToInt64(this string value, int fromBase, out long r
...
+        public static bool IsWindows
         {
             get
             {
-                if (isRunningOnLinux == null)
+                if (!isRunningOnLinux.HasValue)
                 {
-                    int p = (int)Environment.OSVersion.Platform;
-                    isRunningOnLinux = (p == 4) || (p == 6) || (p == 128);
+                    isRunningOnLinux = RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
```
Comment (codeFather2) [inline Sources/PT.PM.Common/CommonUtils.cs]: i think it's bad idea to assign result of `IsOsPlatform(Windows)`, to `isRunningOnLinux`. Also we can have `OSX`. Better to call `IsOsPlatform(OsPlatform.Linux)`

File: Sources/PT.PM.Common/PathUtils.cs [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+                return $@"\\?\{path}";
+            }
+
+            return path.NormalizeDirSeparator();
+        }
+
+        public static string NormalizeDirSeparator(this string path)
+        {
+            string notPlatformSeparator = CommonUtils.IsWindows ? "/" : "\\";
```
Comment (codeFather2) [inline Sources/PT.PM.Common/PathUtils.cs]: i think that `Path.AltDirectorySeparatorChar` property is better for this case, cause it contains '/' for Windows and Mac Os, and '\\' for Linux systems

File: Sources/PT.PM.Common/PathUtils.cs [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+            if (path.Contains(notPlatformSeparator))
```
Comment (codeFather2) [inline Sources/PT.PM.Common/PathUtils.cs]: you shouldn't check for separatorChar contains in path, `Replace` method will make it for you

File: Sources/PT.PM.Common/CommonUtils.cs (hunk 1 above)
Comment (KvanTTT) [reply Sources/PT.PM.Common/CommonUtils.cs]: Yes, it's a misprint.

File: Sources/PT.PM.Common/PathUtils.cs (hunk 3 above)
Comment (KvanTTT) [reply Sources/PT.PM.Common/PathUtils.cs]: I hope this is an optimization, but not sure. It's rather rare case when slashes removed, so I decided to insert such a check. But maybe you are right and it's useless.

File: Sources/PT.PM.Common/PathUtils.cs (hunk 2 above)
Comment (KvanTTT) [reply Sources/PT.PM.Common/PathUtils.cs]: Not correct: `AltDirectorySeparatorChar` is the same as `DirectorySeparatorChar` on Linux, see https://github.com/dotnet/docs/issues/3894

**Pull request merged**